aws_s3_access_grants_boto3_plugin/s3_access_grants_plugin.py

```diff
@@ -74,6 +86,15 @@ def _get_s3_prefix(self, operation_name, request):
                 raise IllegalArgumentException("Source bucket and destination bucket must be the same.")
             prefix_list = [source_split[1], request.context['input_params']['Key']]
             s3_prefix = destination_bucket_name + self._get_common_prefix_for_multiple_prefixes(prefix_list)
+        elif (operation_name == 'ListObjectsV2' or
+              operation_name == 'ListObjects' or
+              operation_name == 'ListObjectVersions' or
+              operation_name == 'ListMultipartUploads'):
+            s3_prefix = request.context['input_params']['Bucket']
+            try:
+                s3_prefix = s3_prefix + "/" + request.context['input_params']['Prefix']
+            except KeyError:
+                pass
```

Comment:
Why do we pass here instead of throwing the error. Correct me if I am wrong, my assumption is that a keyError is expected when there is no Prefix specified i.e. this is a bucket level request?

Reply:
KeyError is when the key is not present. If prefix is not present that means the customer did not add it and only bucket is present. So we can pass.

Reply:
Sounds good. Thanks.